### PR TITLE
fix sending speech.config many times in one TTS session

### PR DESCRIPTION
**Pull request by yulin-li**
Currently, the SDK sends `speech.config` before every audio chunk. This PR fixes this issue.

### CODE REVIEW

File: src/common.speech/SynthesisAdapterBase.ts [hunk 1]
```diff
@@ -146,9 +144,11 @@ export class SynthesisAdapterBase implements IDisposable {
         this.connectionEvents.attach((connectionEvent: ConnectionEvent): void => {
             if (connectionEvent.name === "ConnectionClosedEvent") {
                 const connectionClosedEvent = connectionEvent as ConnectionClosedEvent;
-                this.cancelSynthesisLocal(CancellationReason.Error,
-                    connectionClosedEvent.statusCode === 1007 ? CancellationErrorCode.BadRequestParameters : CancellationErrorCode.ConnectionFailure,
-                    connectionClosedEvent.reason + " websocket error code: " + connectionClosedEvent.statusCode);
+                if (connectionClosedEvent.statusCode !== 1000) {
```
Comment (glharper) [inline src/common.speech/SynthesisAdapterBase.ts:147]: What happens in this flow if the statusCode 1000 is received? We silently ignore the closed event?

File: src/common.speech/SynthesisAdapterBase.ts [hunk 2]
```diff
@@ -216,26 +216,29 @@ export class SynthesisAdapterBase implements IDisposable {
...
 
-            const messageRetrievalPromise = this.receiveMessage();
+        try {
+            await this.connectImpl();
         } catch (e) {
             this.cancelSynthesisLocal(CancellationReason.Error, CancellationErrorCode.ConnectionFailure, e);
             return Promise.reject(e);
         }
+
+        const connection: IConnection = await this.fetchConnection();
```
Comment (glharper) [inline src/common.speech/SynthesisAdapterBase.ts:227]: So this line and the block below are no longer in a try/catch block. What happens if one of the lines throws an exception now?

**Review by glharper — COMMENTED** [review]
See comments

File: src/common.speech/SynthesisAdapterBase.ts (hunk 1 above)
Comment (yulin-li) [reply src/common.speech/SynthesisAdapterBase.ts:147]: Thanks for reviewing. `statusCode` 1000 means normal closure, which is normally caused by client close the connection. The error code info could be found in our [internal wiki ](https://speechwiki.azurewebsites.net/partners/protocol-websockets-tts.html?q=websocket)

File: src/common.speech/SynthesisAdapterBase.ts (hunk 2 above)
Comment (yulin-li) [reply src/common.speech/SynthesisAdapterBase.ts:227]: thx, I will move this into try/catch block.

**Review by glharper — APPROVED** [review]
Approved with addition of try/catch block

**Review by yinhew — COMMENTED** [review]
Is this issue JS specific?

**Comment by yulin-li** [comment]
@yinhew yes, it's JS specific.

**Review by yinhew — APPROVED** [review]

**Pull request merged**